blocks/filter-group/filter-group.js

```diff
@@ -1,42 +1,193 @@
+        // "All" reset. Display all articles.
+        if (selectedFilters.length == 0){
+            article.style.display = '';
+            return;
+        }
+        
+        // Show/hide based on data-tags applied to the article.
+        if (article.dataset?.tags) {
+            const tagsArray = article.dataset.tags.split(',');
+            const hasTag = selectedFilters.some(tag => tagsArray.includes(tag));
+            article.style.display = hasTag ? '' : 'none';
+        }
+    });
+};
+
+/**
+ * Handle click on a filter button from the parent container. Events are bubbled
+ * up from the button. Triggers functionality to update filtered content and
+ * selected state.
+ * @param {PointerEvent} event
+ */
+const handleFilterClick = function (event) {
+    const filterGroup = this;
+    const clickedButton = event.target.closest('.filter-group__button');
+    if (!clickedButton) return;
+
+    // State of button and whether it's the special "All" filter reset.
+    const isSelected = clickedButton.getAttribute('aria-pressed') == 'true';
+    const isAllFilter = clickedButton.matches(':first-of-type');
+
+    if (isAllFilter) {
+        // "All" filter -----
+        // Does nothing if already selected.
+        if (isSelected) return;
+        // Set all other filters as unselected when "All" is selected.
+        filterGroup.querySelectorAll('.filter-group__button').forEach((btn) => {
+            updateFilter(btn, false);
+        });
+        updateFilter(clickedButton, true);
+    } else {
+        // Regular filters -----
+        // Set filter as selected, and deselect "All" if it was selected.
+        const allButton = filterGroup.querySelector('.filter-group__button:first-of-type');
+        updateFilter(clickedButton, !isSelected);
+        
+        // Update state of "All" filter:
+        // - If at least one filter is selected, "All" gets deselected.
+        // - In the case that the only selected filter was just deselected, set "All" back to selected.
+        const hasAnySelectedFilter = document.querySelector('.filter-group__button:not(:first-of-type)[aria-pressed="true"]') != null;
+        updateFilter(allButton, !hasAnySelectedFilter);
+    }
+
+    // Update array of selected filters, and refresh articles displayed on page.
+    const selectedFilters = getCurrentFiltersArray(filterGroup);
+    filterGroup.dataset.selectedFilters = selectedFilters;
+    refreshArticleContent(selectedFilters);
+};
+
+/**
+ * Remove filter buttons for tags that are not assigned to any articles.
+ */
+const removeUnusedTags = async (parentElement) => {
+    const allArticles = await dataStore.getData(dataStore.commonEndpoints.queryIndex);
+    if (allArticles?.data?.length > 0) {
+        parentElement.querySelectorAll('.filter-group__button:not(:first-of-type)').forEach((btn) => {
+            if (!allArticles?.data?.some(article => article?.tag && btn.textContent === article.tag)){
```

Comment:
we could put a `.trim()` on these values to account for whitespace in the documents

Reply:
Good idea. I've added `trim()` where textContent is being accessed.